### DISCUSSION TITLE
Can empty split panes be left open?

### DISCUSSION THREAD

**Discussion by paularmstrong** [zed-industries/zed#14092] [Feature Requests]
This is a feature that Sublime Text has had forever and VSCode took a very long time to implement, but finally got it a few years ago. 

Essentially, I'd like to open an _empty_ split pane (or 3) and leave them open even after all tabs have been closed.

Here's how it looks in VSCode:
![Screenshot 2024-07-10 at 12 46 28](https://github.com/zed-industries/zed/assets/33297/d4f3b343-9cfa-4541-a7d6-a83ae161ac3f)

**Comment by galikhanov** [comment]
I'd like this, too

I always work with 2 files (code and spec) opened next to each other, I use split pane feature for that. The problem is that when i close the last tab the pan disappears. It would not be such a problem, but opening a file in a separate pane takes 3 actions:
- search file and open it in the current pane
- split the view
- close the file in the first pane

**Comment by galikhanov** [comment]
Alternatively it would be nice to have a way to open a file in split pane by shortcut

**Comment by bbb651** [comment]
Turns out it's implemented as a bug 😅
![image](https://github.com/user-attachments/assets/3153515d-7258-4c24-96b3-c9b7c6e80fcb)
When you close files with the edited on disk confirmation, it forgets to do the split closing logic
![image](https://github.com/user-attachments/assets/0bcd1a46-8fac-41e7-9465-a620260f4ce5)

**Comment by aramallo** [comment]
This is the only thing stopping me from using Zed. So I though I would structure the feature request to see if we can bring it to your attention :-)  

### Persistent Multi-Pane Layout with Keyboard Shortcuts

* As a developer using Zed on a wide display,
* I want the ability to quickly split my workspace into multiple panes using keyboard shortcuts,
* So that I can efficiently manage my code across multiple sections and have my layout persist even after restarting the editor.

#### Acceptance Criteria:
1.  Keyboard Shortcuts for Pane Configuration:
    1. On Windows/Linux: Pressing ALT + UP + 2/3/4/5 should create a 2, 3, 4, or 5-pane layout respectively.
    1. On Mac: Pressing OPTION + ⌘ + 2/3/4/5 should achieve the same behavior.
    1. Any number greater than 6 should default to a 5-pane grid.
1.  Pane Persistence:
    1. The configured pane layout must be preserved even after restarting Zed.
    1. When reopening the editor, all panes should appear in their last known state.
1.  Tab Behavior When Adjusting Panes:
    1. Removing a pane should automatically move any open tabs in that pane to one of the remaining panes.
    1. Adding a pane should create an empty pane without redistributing existing tabs.

**Reply by robruder** [reply]
Would love a way to easily move an editor into an already-existing different pane via keyboard shortcuts as well. Have only been poking around for a couple days but haven't found a way to do this that doesn't involve manually dragging a window or splitting into a new pane (which still will leave the editor open in the last pane, and only is helpful on initial open when you don't have already existing panes configured how you'd like)

**Comment by piotrsynowiec** [comment]
It's not the same, but you can right click on the icon to send it to the right dock and this way you can have two docks opened on both sides (one with the files, the other with git changes).

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by dougc84** [reply]
Yay, gotta join yet another Discord server just to submit bug and support requests! Ugh.
